shared/src/ledger/ibc/connection.rs

```diff
@@ -0,0 +1,423 @@
+//! IBC validity predicate for connection module
+
+use std::str::FromStr;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use ibc::ics02_client::client_consensus::{AnyConsensusState, ConsensusState};
+use ibc::ics02_client::client_state::AnyClientState;
+use ibc::ics02_client::context::ClientReader;
+use ibc::ics02_client::height::Height;
+use ibc::ics03_connection::connection::{ConnectionEnd, Counterparty, State};
+use ibc::ics03_connection::context::ConnectionReader;
+use ibc::ics03_connection::handler::verify::verify_proofs;
+use ibc::ics07_tendermint::consensus_state::ConsensusState as TendermintConsensusState;
+use ibc::ics23_commitment::commitment::CommitmentPrefix;
+use ibc::ics24_host::identifier::{ClientId, ConnectionId};
+use ibc::ics24_host::Path;
+use ibc::proofs::Proofs;
+use tendermint_proto::Protobuf;
+
+use super::{Error, Ibc, Result, StateChange};
+use crate::ledger::storage::{self, StorageHasher};
+use crate::types::address::{Address, InternalAddress};
+use crate::types::ibc::{
+    ConnectionOpenAckData, ConnectionOpenConfirmData, ConnectionOpenTryData,
+    Error as IbcDataError,
+};
+use crate::types::storage::{BlockHeight, Epoch, Key, KeySeg};
+
+pub(super) const COUNTER_PATH: &str = "connections/counter";
```

Comment:
maybe we could add a function to the storage key with the
```
let path = COUNTER_PATH.to_owned();
Key::ibc_key(path).expect("Creating a key for a connection counter failed");
```

Reply:
Thank you for your suggestion!
I added the function to the storage key.